libmamba/src/core/curl.hpp

```diff
@@ -64,6 +64,38 @@ namespace mamba
         bool m_serious;
     };
 
+    class CURLId
```

Comment:
Did you consider making this type definition header-only/inline? It seems to be a good candidate.